### PR TITLE
fix the exif_read_data issue

### LINKED ISSUE #34958: [Bug]: Error: exif_read_data(): File not supported at /var/www/nextcloud/lib/private/Metadata/Provider/ExifProvider.php#59
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Upgraded from NC 24.0.7 to NC 25.0.1

### Steps to reproduce

N/A

### Expected behavior

N/A

### Installation method

Community Manual installation with Archive

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
N/A
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

```shell
{"reqId":"5pmQTuYNkJWS2c7Wq3i7","level":3,"time":"2022-11-03T12:02:55-05:00","remoteAddr":"X.X.X.X","user":"admin","app":"PHP","method":"PUT","url":"/remote.php/dav/files/ixheli/Photos/2022/11/22-11-03%2010-10-00%201115.png","message":"exif_read_data(): File not supported at /var/www/nextcloud/lib/private/Metadata/Provider/ExifProvider.php#59","userAgent":"Mozilla/5.0 (iOS) Nextcloud-iOS/4.5.3","version":"25.0.1.1","exception":{"Exception":"Error","Message":"exif_read_data(): File not supported at /var/www/nextcloud/lib/private/Metadata/Provider/ExifProvider.php#59","Code":0,"Trace":[{"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/lib/private/Metadata/Provider/ExifProvider.php","line":59,"function":"exif_read_data"},{"file":"/var/www/nextcloud/lib/private/Metadata/MetadataManager.php","line":68,"function":"execute","class":"OC\\Metadata\\Provider\\ExifProvider","type":"->"},{"file":"/var/www/nextcloud/lib/private/Metadata/FileEventListener.php","line":101,"function":"generateMetadata","class":"OC\\Metadata\\MetadataManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/ServiceEventListener.php","line":87,"function":"handle","class":"OC\\Metadata\\FileEventListener","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":251,"function":"__invoke","class":"OC\\EventDispatcher\\ServiceEventListener","type":"->"},{"file":"/var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php","line":73,"function":"callListeners","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":88,"function":"dispatch","class":"Symfony\\Component\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php","line":100,"function":"dispatch","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Node/HookConnector.php","line":118,"function":"dispatchTyped","class":"OC\\EventDispatcher\\EventDispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Hook.php","line":106,"function":"postWrite","class":"OC\\Files\\Node\\HookConnector","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/File.php","line":471,"function":"emit","class":"OC_Hook","type":"::"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/File.php","line":398,"function":"emitPostHooks","class":"OCA\\DAV\\Connector\\Sabre\\File","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php","line":151,"function":"put","class":"OCA\\DAV\\Connector\\Sabre\\File","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1098,"function":"createFile","class":"OCA\\DAV\\Connector\\Sabre\\Directory","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":504,"function":"createFile","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPut","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Server.php","line":360,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/remote.php","line":171,"args":["/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"636408dd7808c"}
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by szaimen**
Fix https://github.com/nextcloud/server/issues/34958

### CODE REVIEW

File: lib/private/Metadata/Provider/ExifProvider.php [hunk 1]
```diff
@@ -56,10 +56,10 @@ public function execute(File $file): array {
 			// But I don't understand why 1 as a special meaning.
 			// Revert right after reading the exif data.
 			$oldBufferSize = stream_set_chunk_size($fileDescriptor, 1);
-			$data = exif_read_data($fileDescriptor, 'ANY_TAG', true);
+			$data = @exif_read_data($fileDescriptor, 'ANY_TAG', true);
```
Comment (szaimen) [inline lib/private/Metadata/Provider/ExifProvider.php:59]: this was actually the only way how I could suppress the log message by testing with https://user-images.githubusercontent.com/12234510/205661916-3e7b14b4-6d4b-4dc2-9903-5265f95a2aee.png

Comment (DrNanoLan89) [reply lib/private/Metadata/Provider/ExifProvider.php:59]: Ok.

**Comment by simonspa** [comment]
What I find odd is that this seems to be only triggered by Android client uploads, but not when adding the exact same file via web interface. In one case, the error is thrown, in the other case EXIF data is retrieved just fine - so I guess there is still something else going on.

Nevertheless, suppressing this error seems to make sense to me.

**Review by simonspa — APPROVED** [review]

**Review by artonge — COMMENTED** [review]
Do you understand why the try/catch is not enough ?
Not a fan of silencing error like that.
Maybe changing `} catch (\Exception $ex) {` for `} catch (\Throwable $ex) {` and changing log level to info would be enough ?

**Comment by come-nc** [comment]
> Do you understand why the try/catch is not enough ? Not a fan of silencing error like that. Maybe changing `} catch (\Exception $ex) {` for `} catch (\Throwable $ex) {` and changing log level to info would be enough ?

https://www.php.net/manual/function.exif-read-data.php states:
> Errors of level E_WARNING and/or E_NOTICE may be raised for unsupported tags or other potential error conditions, but the function still tries to read all comprehensible information.

You cannot try/catch those errors, that goes straight to log.
The only way to catch them is to override the logger before the call, which is not practical.
From what I understand of the documentation, it does not throw because in some cases it still returns partial information.

**Comment by szaimen** [comment]
Thanks @come-nc! So good to merge?

**Review by artonge — APPROVED** [review]

**Pull request merged**